### DISCUSSION TITLE
2.3.6 Home Assistant MQTT tweaks

### DISCUSSION THREAD

**Discussion by jmp2000** [aqualinkd/AqualinkD#310] [Q&A]
Here's the latest on 2.3.6 running 64-bit on a Pi Zero 2 W.   A couple fine-tuning questions in Home Assistant (MQTT default view)...

1. Air/Pool/Spa temperatures are 'correct in C', but 'displaying in F'.   Any thoughts on what to adjust there?

2. SWG Messages seems to always say 'generating' even when the pool pump or saltwater generator is off.

3. In the 'thermostats' - for Freeze Protection/Pool Heater/SWG/Spa Heater.   I don't have a Pool Heater installed.  Any tips on how to 'remove that 2nd thermostat'?

4. Similar to question 1, on all of the 'thermostats' - it's the correct temp in C, but displaying as F.   Whatever fixes item 1, will likely fix this one as well.

Amazing stuff... really impressed that I now have it all going through HA... going to make a big difference on ease of use!!

![Screenshot 1](https://github.com/sfeakes/AqualinkD/assets/149847986/33c7fe26-3e7f-4d7c-808e-ce92946026fb)
![Screenshot 2](https://github.com/sfeakes/AqualinkD/assets/149847986/5cb7401d-ff24-402e-a717-9be5760baef9)

**Comment by sfeakes** [comment]
#1 you have MQTT set to the deg C.  https://github.com/sfeakes/AqualinkD/wiki#HASSIO   Check the convert_mqtt_temp_to_c setting in aqualinkd.conf.
#2 I’ll check that in code, but you may want to also check ‘ read_RS485_swg = yes’ is in your config.
#3 you should set that specific button to NONE in the config.  But I’ll need to check if AqualinkD follows that setting for home assistant.
#4 same as #1.

**Reply by jmp2000** [reply]
You're right on #1 and #4 - I incorrectly had that setting set to "yes".   Changed it to "no" and everything went back to F properly.

For 2 - I have that code in there, but it shows generating always in HA.
For 3 - I have NONE in the bottom configuration section of aqualink.conf, but it still shows that 'thermostat' for the pool heater that doesn't exist (at least it shows that it has a correct pool temp, in F! )

**Reply by jmp2000** [reply]
Now here's the real headscratcher on the SWG.   It will work fine for a day.   Today (2nd day I turned on the pool pump in the evening) - the SWG will not turn on.   Turned on/off the pump several times; SWG never starts.   Only after I power off the Pi Zero 2 W, and then restart the pool pump from the One Touch panel.... then the SWG turns on with the pool pump?!?!    It's like the signal will turn on the SWG "once", and then it never turns off again... I wonder if this is somehow connected to Item #2 above??

**Reply by sfeakes** [reply]
These two could be linked, I did make a small change in SWG in the last release. Previously it would pick up SWG info from both panel and SWG directly, latest release I changes it to one or other.  Can you try changing  ‘ read_RS485_swg ’ to ‘no’, and see if you get the expected result.

**Comment by sfeakes** [comment]
For #3 it’s probably something I didn’t think about when coding the HA stuff, I’ll fix in next release.
For #2, what does the aqualinkd web ui state?  And in HA, does it say the SWG % is 0 when it’s off?  

  I’m wondering if your SWG is wired to be constantly on / powered, most will turn off when set to 0 to save the PSU.
